depdive/src/github.rs

```diff
@@ -165,20 +175,52 @@ impl GitHubAnalyzer {
             .unwrap_or(false)
     }
 
-    fn get_github_repo_fullname(repo_url: &Url) -> Result<String> {
+    fn get_github_repo_fullname(&self, repo_url: &Url) -> Result<String> {
```

Comment:
Change this to return `Result<String, InvalidGitHubRepoError>` so that you don't have to downcast above.